db/change_listener.go

```diff
@@ -181,22 +181,33 @@ func (listener changeListener) TapFeed() base.TapFeed {
 //////// NOTIFICATIONS:
 
 // Changes the counter, notifying waiting clients.
-func (listener *changeListener) Notify(keys base.Set) {
+func (listener *changeListener) Notify(keys channels.Set) {
 
 	if len(keys) == 0 {
 		return
 	}
 	listener.tapNotifier.L.Lock()
 	listener.counter++
 	for key := range keys {
-		listener.keyCounts[key] = listener.counter
+		listener.keyCounts[key.String()] = listener.counter
 	}
 	base.DebugfCtx(context.TODO(), base.KeyChanges, "Notifying that %q changed (keys=%q) count=%d",
 		base.MD(listener.bucketName), base.UD(keys), listener.counter)
 	listener.tapNotifier.Broadcast()
 	listener.tapNotifier.L.Unlock()
 }
 
+// Changes the counter, notifying waiting clients. Only use for single update is updating user, which might be collection scoped in the future.
+func (listener *changeListener) notifyUser(key string) {
```

Comment:
I like the idea of having a separate Notify for a single key, since that's currently how principals are notified.  I don't think there's anything in here that's specific to user/role, though.  Might be better to just switch to NotifyKey() and maybe NotifyChannels for the version that accepts channels.Set?

Reply:
Fixed.